### ISSUE TITLE
Change the "reduction" of Pytorch Losses

### ISSUE THREAD

**Issue by Eric-Wallace** [pytorch/xla#1271]
## 🚀 Feature
 Many torch loss functions, e.g., cross entropy, nll loss, etc., have an ability to change the `reduce` from mean to other values. This is used in multiple places inside Fairseq, where we would like to not take the mean and instead use the sum or not reduce the loss at all. This is especiallyuseful for changing the normalization / doing masking. For example, here https://github.com/pytorch/fairseq/blob/master/fairseq/criterions/masked_lm.py#L54

Currently, the loss functions are supported in XLA, but setting the reduce flags is unsupported and the objects move to the CPU (i.e., I see `aten:` calls for the cross entropy). This slows speeds significantly 

In the meantime, if your end goal is to have a `sum` instead of a `mean`, you can simply call the regular loss and then multiply by the number of elements. However, there are also cases where we need `none` reduction, which there is no work around that I am aware of.

**Comment by dlibenzi** [comment]
PR #1282 adds all reduction types and weight support to `nll_loss`.
You mentioned that you see `aten::*` with Cross Entropy Loss, but I cannot with #1282 applied, so the ones you were seeing must have come from the partial handling of `nll_loss`.

```Python
import torch                                                                                                                                                                                                                                
import torch_xla                                                                                                                                                                                                                            
import torch_xla.core.xla_model as xm                                                                                                                                                                                                       
import torch_xla.debug.metrics as met                                                                                                                                                                                                       
import torch.nn as nn                                                                                                                                                                                                                       
from torch.autograd import Variable                                                                                                                                                                                                         
                                                                                                                                                                                                                                            
device = xm.xla_device()                                                                                                                                                                                                                    
output = torch.FloatTensor([0, 0, 0, 1]).to(device).view(1, -1).requires_grad_()                                                                                                                                                            
target = torch.LongTensor([3]).to(device)                                                                                                                                                                                                   
                                                                                                                                                                                                                                            
criterion = nn.CrossEntropyLoss()                                                                                                                                                                                                           
loss = criterion(output, target)                                                                                                                                                                                                            
print(loss.item())                                                                                                                                                                                                                          
loss.backward()                                                                                                                                                                                                                             
print(met.metrics_report())
```

```
Counter: aten::_local_scalar_dense
  Value: 1
Counter: xla::_log_softmax
  Value: 1
Counter: xla::_log_softmax_backward_data
  Value: 1
Counter: xla::copy_
  Value: 2
Counter: xla::empty
  Value: 2
Counter: xla::full
  Value: 2
Counter: xla::full_like
  Value: 1
Counter: xla::nll_loss
  Value: 1
Counter: xla::nll_loss_backward
  Value: 1
Counter: xla::nll_loss_forward
  Value: 1
Counter: xla::ones_like
  Value: 1
Counter: xla::view
  Value: 1
```

The `aten::_local_scalar_dense` is the one `item()` call.

**Comment by Eric-Wallace** [comment]
Hi @dlibenzi this is working great!

Sorry to be that guy, but is it also possible to add this for binary_cross_entropy loss? I am also using that with reduction = 'none"

**Comment by dlibenzi** [comment]
We will take a look.
For `we`, I mean @ailzhang 😁
